cmd/influx2cortex/main.go

```diff
@@ -5,51 +5,76 @@ import (
 	"errors"
 	"flag"
 	"fmt"
+	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
 	"os"
 
 	"github.com/grafana/dskit/flagext"
 	"github.com/grafana/dskit/services"
 	"github.com/grafana/influx2cortex/pkg/influx"
+	"github.com/grafana/influx2cortex/pkg/internalserver"
 	"github.com/weaveworks/common/logging"
 	"github.com/weaveworks/common/signals"
 )
 
 func main() {
-	conf := influx.ProxyConfig{}
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stdout))
+	promRegisterer := prometheus.DefaultRegisterer
+
+	proxyConfig := influx.ProxyConfig{
+		Logger:     logger,
+		Registerer: promRegisterer,
+	}
+	internalServerConfig := internalserver.ServiceConfig{}
 
-	flag.BoolVar(&conf.EnableAuth, "auth.enable", true, "require X-Scope-OrgId header")
 	flagext.RegisterFlags(
-		&conf.HTTPConfig,
-		&conf.RemoteWriteConfig,
+		&proxyConfig,
+		&internalServerConfig,
 	)
 	flag.Parse()
 
-	service, err := influx.NewProxy(conf)
+	appServices := make([]services.Service, 0)
+
+	proxyService, err := influx.NewProxy(proxyConfig)
 	if err != nil {
 		_, _ = fmt.Fprintf(os.Stderr, "error instantiating influx2cortex proxy: %s\n", err)
 		os.Exit(1)
 	}
+	appServices = append(appServices, proxyService)
 
-	servCtx, cancelFn := context.WithCancel(context.Background())
-	defer cancelFn()
-
-	err = services.StartAndAwaitRunning(servCtx, service)
+	internalService, err := internalserver.NewService(internalServerConfig, proxyService.Logger)
```

Comment:
I think we should use the `logger` variable rather than the `proxyService.Logger`. `proxyService.Logger` might have some proxy-specific configs for the logger that you wouldn't want in the logger for internal server. I don't think there's an actual issue right now but in general it's something I think we should avoid.

Reply:
Whoop, nice catch - I refactored that `logger` definition out precisely so there wasn't this weird dependency on another service's logger; I guess I forgot to go back and actually make the change after that refactor 😅